cmd/syncer/main.go

```diff
@@ -122,13 +132,13 @@ func main() {
 // initSyncerComponents initializes syncer components that are dependant on the leader election algorithm.
 // This function is only called by the leader instance of vsphere-syncer, if enabled.
 // TODO: Change name from initSyncerComponents to init<Name>Components where <Name> will be the name of this container
-func initSyncerComponents(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor, configInfo *types.ConfigInfo) func(ctx context.Context) {
+func initSyncerComponents(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor, configInfo *types.ConfigInfo, coInitParams *interface{}) func(ctx context.Context) {
```

Comment:
Why do we have pass this as an extra parameter to `initSyncerComponents`?
Isn't `COInitParams` already part of Syncer package and a global var

Reply:
I do not want to import `syncer` package in other packages like `admissionhandler` and `Storagepool`. Therefore I created a new global variable in `admissionhandler` and assigning the value in syncer/main.go. For storagepool, I am passing it as an argument.